internal/operators/ocs/custom_manifest.go

```diff
@@ -0,0 +1,82 @@
+package ocs
+
+import (
+	"bytes"
+	"os"
+	"text/template"
+)
+
+// Subscription Manifest for testing internal builds
+func ocsCustomSubscription() (string, error) {
+	data := map[string]string{
+		"OPERATOR_NAMESPACE":         Operator.Namespace,
+		"OPERATOR_SUBSCRIPTION_NAME": Operator.SubscriptionName,
+	}
+
+	const ocsSubscription = `apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: "{{.OPERATOR_SUBSCRIPTION_NAME}}"
+  namespace: "{{.OPERATOR_NAMESPACE}}"
+spec:
+  installPlanApproval: Automatic
+  name: ocs-operator
+  source: ocs-catalogsource
+  sourceNamespace: openshift-marketplace`
+
+	tmpl, err := template.New("ocsSubscription").Parse(ocsSubscription)
+	if err != nil {
+		return "", err
+	}
+	buf := &bytes.Buffer{}
+	err = tmpl.Execute(buf, data)
+	if err != nil {
+		return "", err
+	}
+	return buf.String(), nil
+}
+
+// CatalogSource Manifest for testing internal builds
+func ocsCatalogSource() (string, error) {
+
+	image := "quay.io/rhceph-dev/ocs-registry:latest"
+	if os.Getenv("OCS_TEST_IMAGE") != "" {
+		image = os.Getenv("OCS_TEST_IMAGE")
+	}
+	data := map[string]string{
+		"OPERATOR_IMAGE": image,
+	}
+
+	const CatalogSource = `apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: ocs-catalogsource
+  namespace: openshift-marketplace
+  labels:
+      ocs-operator-internal: "true"
+spec:
+  displayName: Openshift Container Storage
+  icon:
+    base64data: ""
+    mediatype: ""
+  image: "{{.OPERATOR_IMAGE}}"
+  publisher: Red Hat
+  sourceType: grpc
+  # If the registry image still have the same tag (latest-stable-4.6, or for stage testing)
+  # we need to have this updateStrategy, otherwise we will not see new pushed content.
+  updateStrategy:
+    registryPoll:
+      interval: 15m`
```

Comment:
This part can be maybe deleted from here.
It has its pros and cons.

Pros: It keeps you cluster upgraded when new stable build is produced and you are using such tag like latest-stable-4.6.

But if you are in the middle of the run - you maybe don't want to have cluster upgraded in the middle - for this purpose user should rather use some static tag like:  4.8.0-414.ci

I think you can remove it from here probably. For manual re-load of the catalog, user can also delete the pod for ocs-catalogsource from openshift-marketplace.

Reply:
updated!